node/src/Unison/ABT/Extra.hs

```diff
@@ -21,10 +21,12 @@ import qualified Data.Map as Map
 import qualified Data.Vector as Vector
 import qualified Unison.Digest as Digest
 
-hash :: forall f . (Foldable f, Digest.Digestable1 f) => Term f -> Digest.Hash
+-- | We ignore annotations in the `Term`, as these should never affect the
+-- meaning of the term.
```

Comment:
We rely more heavily on the type-checking algorithm to give meaning to our programs if we do this. Part of me would rather have the fully-elaborated AST be the "final truth" of the program that we persist. Text-based software development problem *doesn't* do this today because it would make refactoring that much harder, but the ideas you propose in your last post may alleviate this. Furthermore we can distinguish inferred and explicit types if we need.